@navikt/core/react/src/overlays/overlay/hooks/useRefWithInit.ts

```diff
@@ -0,0 +1,25 @@
+"use client";
+
+import React, { useRef } from "react";
+
+const UNINITIALIZED = {};
+
+/**
+ * useRef initialized with a function on mount.
+ */
+function useRefWithInit<T>(init: () => T): React.RefObject<T>;
+function useRefWithInit<T, U>(
+  init: (arg: U) => T,
+  initArg: U,
+): React.RefObject<T>;
```

Comment:
I think `MutableRefObject` would be more correct here. Then you don't have to use `!` everywhere.
```suggestion
function useRefWithInit<T>(init: () => T): React.MutableRefObject<T>;
function useRefWithInit<T, U>(
  init: (arg: U) => T,
  initArg: U,
): React.MutableRefObject<T>;
```

Reply:
Could do that, but this is currently just a quirk with React < 19. In 19 its fixed so we don't need `!` anywhere 🎉 
Tested with react 19 types and getting this then `@deprecated — Use RefObject instead.`